### PR TITLE
Feature/jibreen order status

### PR DESCRIPTION
**Pull request by JbreenJboor1212**
**The code defines CRUD endpoints for managing order statuses in FastAPI. It includes the following:**

- **Create Status `(POST /)`**: Adds a new order status if the name does not already exist, returning the created status.
- **Get Status by ID `(GET /{status_id})`**: Retrieves the order status by its unique ID, returning a 404 error if not found.
- **Update Status `(PUT /{status_id})`**: Updates the name of an existing status if the new name is unique, otherwise returns a conflict error.
- **Delete Status `(DELETE /{status_id})`**: Deletes an order status if it is not linked to any existing orders, otherwise prevents deletion.

All endpoints ensure that only an authenticated admin user can access or modify the statuses, using proper error handling to catch potential issues.

### CODE REVIEW

**Review by Habeeb-Awawda — APPROVED** [review]

File: app/models.py [hunk 1]
```diff
@@ -73,4 +73,31 @@ class Config:
...
         
-    
\ No newline at end of file
+
+# ------------ Status Model -----------------#
+# Models
+class OrderStatusBaseModel(BaseModel):
+    name: str = Field(..., examples=["Pending", "Processing", "Completed", "Canceled"],description="Status name, e.g. Pending, Processing, Completed, Canceled")
+
+class OrderStatusCreateModel(OrderStatusBaseModel):
```
Comment (Mohammed-samaheen) [inline app/models.py]: please follow this convention of naming models 
[method][model type][request tupe]Model
so this should be
CreateOrderStatusRequestModel

File: app/api/routes/status.py [hunk 2]
```diff
@@ -1,8 +1,68 @@
...
+@router.post("/", response_model=OrderStatusResponseModel, status_code=status.HTTP_201_CREATED)
+async def create_status(status: OrderStatusCreateModel, current_user: User = Depends(get_current_admin_user)):
+   
+    if any(existing_status["name"].lower() == status.name.lower() for existing_status in statusOrders_db.values()):
+        raise HTTPException(status_code=400, detail="Status name already exists")
+    
+    new_status = OrderStatusModel(
+        name=status.name,
+        created_at=datetime.now(timezone.utc),
+        updated_at=datetime.now(timezone.utc)
```
Comment (Mohammed-samaheen) [inline app/api/routes/status.py]: update date for created status?

File: app/models.py [hunk 3]
```diff
@@ -73,4 +73,31 @@ class Config:
...
+
+class OrderStatusResponseModel(OrderStatusModel):
+    pass
+
+    class Config:
+        json_encoders = {
+            datetime: lambda v: v.strftime('%Y-%m-%d %H:%M:%S')
+        }
+        
+class DeleteResponseModel(BaseModel):
```
Comment (Mohammed-samaheen) [inline app/models.py]: Please make sure to follow the task description. In delete, we return nothing.

File: app/models.py [hunk 4]
```diff
@@ -73,4 +73,31 @@ class Config:
...
+
+class OrderStatusUpdateModel(OrderStatusBaseModel):
+    pass
+
+class OrderStatusModel(OrderStatusBaseModel):
+    id: UUID = Field(default_factory=uuid4)
+    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    updated_at: Optional[datetime] = Field(default=None)
+
+class OrderStatusResponseModel(OrderStatusModel):
```
Comment (Mohammed-samaheen) [inline app/models.py]: create doesn't have updated_at

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

File: app/models.py (hunk 1 above)
Comment (JbreenJboor1212) [reply app/models.py]: I forgot about that, I'll refactor it now.

File: app/api/routes/status.py (hunk 2 above)
Comment (JbreenJboor1212) [reply app/api/routes/status.py]: I think when creating the first one, the same will be updated. But if it doesn't work, I'll return null.

File: app/models.py (hunk 3 above)
Comment (JbreenJboor1212) [reply app/models.py]: I think when creating the first one, the same will be updated. But if it doesn't work, I'll return null.

File: app/models.py (hunk 1 above)
Comment (JbreenJboor1212) [reply app/models.py]: Done. Your idea is better than mine. Thanks!

File: app/api/routes/status.py (hunk 2 above)
Comment (JbreenJboor1212) [reply app/api/routes/status.py]: Done. Your idea is better than mine. Thanks!

File: app/models.py (hunk 3 above)
Comment (JbreenJboor1212) [reply app/models.py]: Done. Your idea is better than mine. Thanks!

File: app/models.py (hunk 4 above)
Comment (JbreenJboor1212) [reply app/models.py]: Done. Your idea is better than mine. Thanks!

Comment (JbreenJboor1212) [reply app/models.py]: @Mohammed-samaheen Please review my changes.

**Review by Mohammed-samaheen — APPROVED** [review]

**Pull request merged**